architecture.md

```diff
@@ -0,0 +1,65 @@
+- Multiple Individual Subgraphs Benefits:
+  - Makes it easier for users to fork the subgraph to add entities.
+  - It will sync way faster.
+  - It will allow a subgraph developer to come along and think only in the perspective of their subgraph.
+- Single Subgraph Combined:
+  - An unexplored opportunity in The Graph is cross-protocol Subgraphs. It highlights to power of Subgraphs as Open APIs. A Subgraph can be an API into a single protocol, a single account, or multiple protocols, or anything else you can think of. However, 95% of people only think of Subgraphs as a view into a single protocol - and this is likely due to how early we still are. Most subgraphs are purpose built for front ends, created by a single dev or dev team. This would provide awareness and marketing for The Graph / Subgraphs.
+  - A general subgraph that can aggregate all lending data is a beautiful thing. In data science there is an term EXTRACT, TRANSFORM & LOAD (ETL). A general subgraph removes the EXTRACT and TRANSFORM and leaves just the LOAD (i.e. writing queries and loading into their DB) for a data analyst to do.
+  - It is likely that a Subgraph of this size will take +2 weeks to sync. But when the [Fire Hose](https://github.com/streamingfast/firehose) upgrade is ready, this would be a perfect subgraph to test the limits of the indexing speed. 
+  - We should be thinking about this in terms of the The Graph Decentralized Network. It is good for multiple reasons:
+    - Core teams benefit by being able to test the economics of query fees on the decentralized network, test the indexing speed against a massive subgraph, and marketing for the protocol at large.
+    - Indexers lives are easier as they only need to index 1 Subgraph. More queries for a single subgraph. Less transactions and mental overhead for the indexers (ETH is expensive for them to stake on a subgraph).
+    - It makes curators lives easier as well as they need to discover less subgraphs.
+
+The script should simply be a typescript file which combines all manifests into a single manifest, and deploys the subgraph. It will be executed with a node script in `package.json`.
+
+Note - ultimately, as one big subgraph or as many small subgraphs - it is a decision of how to split up the indexing work. Example:
+- If you create one big subgraph, you do all the indexing work ahead of time, and you require the Subgraph Developer to build the Subgraph to include all protocols.
+- If you create many small subgraphs, you parallelize the indexing, but push the work out to Indexers, Curators, and Data Analysts to determine how to aggregate all the data. 
+
+In both cases, queries must be written by the consumers in the end. Neither is right or wrong. What ultimately decides the usefulness of a Subgraph is how many queries it receives in the decentralized network.
+
+## What should the base schema be for the lending protocols?
```

Comment:
We can start out with lending protocols but we should take into account of other protocol categories as well when designing

Reply:
Yeah, even though a dex subgraph would be different, we should try to make fields as consistent as possible, and consider names that are consistent.